crates/core/src/db/datastore/traits.rs

```diff
@@ -48,20 +47,43 @@ pub trait DataRow: Send + Sync {
 
     fn view_product_value<'a>(&self, data_ref: Self::DataRef<'a>) -> &'a ProductValue;
 }
+/// `ReadTx` Trait
+///
+/// Provides common methods for both read and write transaction types.
```

Comment:
I'm almost certain that there should be no common methods for read and write transaction types, so I'm not sure what these means.

Reply:
It is indeed for common methods between transaction types, I am not sure why we shouldn't have them.
This is for things both type of tx makes sense like Iters